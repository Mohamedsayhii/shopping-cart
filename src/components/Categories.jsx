import { Link } from 'react-router-dom';
import styled from 'styled-components';

const Wrapper = styled.div`
	background-image: url('/Casa.jpg');
	display: grid;
	width: 100%;
	grid-template-columns: 1fr 1fr 1fr;
	gap: 1rem;
	height: 92vh;
	padding: 2rem;
	place-items: center;

	.category {
		display: flex;
		flex-direction: column;
		align-items: center;
		background-color: green;

		img {
			width: 350px;
			height: 500px;
			object-fit: fill;
		}

		h1 {
			color: white;
			padding: 1rem 0rem;
		}
	}

	@media screen and (max-width: 1100px) {
		.category {
			img {
				width: 260px;
				height: 400px;
			}
		}
	}

	@media screen and (max-width: 800px) {
		.category {
			img {
				width: 190px;
				height: 300px;
			}
		}
	}

	@media screen and (max-width: 600px) {
		height: auto;
		grid-template-columns: 1fr;
		grid-template-rows: 1fr 1fr 1fr;

		.category {
			img {
				width: 200px;
				height: 300px;
			}
		}
	}
`;

function Categories() {
	return (
		<Wrapper>
			<Link to='men'>
				<div className='category'>
					<img src='/Men.jpg' alt='' />
					<h1>Men</h1>
				</div>
			</Link>
			<Link to='women'>
				<div className='category'>
					<img src='/Women.jpg' alt='' />
					<h1>Women</h1>
				</div>
			</Link>
			<Link to='jewelry'>
				<div className='category'>
					<img src='/Jewelry.jpg' alt='' />
					<h1>Jewelry</h1>
				</div>
			</Link>
		</Wrapper>
	);
}

export default Categories;
