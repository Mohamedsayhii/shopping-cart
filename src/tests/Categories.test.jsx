import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import Categories from '../components/Categories';
import { MemoryRouter } from 'react-router-dom';

describe('testing categories components', () => {
	it('should render 3 categories', () => {
		const { container } = render(
			<MemoryRouter>
				<Categories />
			</MemoryRouter>
		);

		expect(container.getElementsByClassName('category').length).toBe(3);
	});
});
