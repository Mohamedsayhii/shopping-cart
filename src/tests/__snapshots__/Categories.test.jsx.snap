// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing categories components > should render 3 categories 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-blHHSb fUSza-d"
      >
        <a
          href="/men"
        >
          <div
            class="category"
          >
            <img
              alt=""
              src="/Men.jpg"
            />
            <h1>
              Men
            </h1>
          </div>
        </a>
        <a
          href="/women"
        >
          <div
            class="category"
          >
            <img
              alt=""
              src="/Women.jpg"
            />
            <h1>
              Women
            </h1>
          </div>
        </a>
        <a
          href="/jewelry"
        >
          <div
            class="category"
          >
            <img
              alt=""
              src="/Jewelry.jpg"
            />
            <h1>
              Jewelry
            </h1>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-blHHSb fUSza-d"
    >
      <a
        href="/men"
      >
        <div
          class="category"
        >
          <img
            alt=""
            src="/Men.jpg"
          />
          <h1>
            Men
          </h1>
        </div>
      </a>
      <a
        href="/women"
      >
        <div
          class="category"
        >
          <img
            alt=""
            src="/Women.jpg"
          />
          <h1>
            Women
          </h1>
        </div>
      </a>
      <a
        href="/jewelry"
      >
        <div
          class="category"
        >
          <img
            alt=""
            src="/Jewelry.jpg"
          />
          <h1>
            Jewelry
          </h1>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
